### PR TITLE
python-connector-base: patch vulnerabilities

### LINKED ISSUE #41060: Patch python-connector-base:1.2.2 vulnerabilities
We currently have [8 vulnerabilities](https://airbyte.metabaseapp.com/question/9978-latest-python-base-image-vulnerabilities?base_image_address=docker.io%2Fairbyte%2Fpython-connector-base%3A1.2.2%40sha256%3A57703de3b4c4204bd68a7b13c9300f8e03c0189bffddaffc796f1da25d2dbea0) on our latest python connector base image version.

We should patch them before releasing new base image versions as it will guarantee that connectors that can't upgrade to a new major base image are secure.

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte/issues/41060
Patch python-connector-base image vulnerabilities.

## How
* Using the latest Python 3.9.19 root image
* Update system dependencies

### Before (`1.2.2`)
```bash
➜  ~ grype docker.io/airbyte/python-connector-base:1.2.2 --only-fixed
 ✔ Vulnerability DB                [no update available]
 ✔ Pulled image
 ✔ Loaded image                                                                                                                                               index.docker.io/airbyte/python-connector-base:1.2.2
 ✔ Parsed image                                                                                                                           sha256:2ac5cd4d8bb4805ffd829e7a57fd11ce82bb0e7a2be6d1f336d17dd0fb852990
 ✔ Cataloged contents                                                                                                                            97027dfab14c2d167c5bd95754ae2d707f4e2c116c1183dfa220d5d817a49e55
   ├── ✔ Packages                        [234 packages]
   ├── ✔ File digests                    [3,728 files]
   ├── ✔ File metadata                   [3,728 locations]
   └── ✔ Executables                     [966 executables]
 ✔ Scanned for vulnerabilities     [27 vulnerability matches]
   ├── by severity: 3 critical, 18 high, 39 medium, 14 low, 115 negligible (41 unknown)
   └── by status:   27 fixed, 203 not-fixed, 203 ignored
NAME              INSTALLED          FIXED-IN           TYPE    VULNERABILITY        SEVERITY
certifi           2024.6.2           2024.07.04         python  GHSA-248v-346w-9cwc  Low
libarchive13      3.6.2-1            3.6.2-1+deb12u1    deb     CVE-2024-26256       High
libcurl4          7.88.1-10+deb12u5  7.88.1-10+deb12u6  deb     CVE-2024-2398        High
libcurl4          7.88.1-10+deb12u5  7.88.1-10+deb12u6  deb     CVE-2024-2004        Low
libgnutls30       3.7.9-2+deb12u2    3.7.9-2+deb12u3    deb     CVE-2024-28835       Medium
libgnutls30       3.7.9-2+deb12u2    3.7.9-2+deb12u3    deb     CVE-2024-28834       Medium
libgssapi-krb5-2  1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37371       Unknown
libgssapi-krb5-2  1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37370       Unknown
libk5crypto3      1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37371       Unknown
libk5crypto3      1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37370       Unknown
libkrb5-3         1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37371       Unknown
libkrb5-3         1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37370       Unknown
libkrb5support0   1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37371       Unknown
libkrb5support0   1.20.1-2+deb12u1   1.20.1-2+deb12u2   deb     CVE-2024-37370       Unknown
libssl3           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2024-0727        Medium
libssl3           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2023-6129        Medium
libssl3           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2023-5678        Medium
libssl3           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2023-6237        Unknown
libsystemd0       252.22-1~deb12u1   252.23-1~deb12u1   deb     CVE-2023-50387       High
libsystemd0       252.22-1~deb12u1   252.23-1~deb12u1   deb     CVE-2023-50868       Unknown
libudev1          252.22-1~deb12u1   252.23-1~deb12u1   deb     CVE-2023-50387       High
libudev1          252.22-1~deb12u1   252.23-1~deb12u1   deb     CVE-2023-50868       Unknown
openssl           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2024-0727        Medium
openssl           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2023-6129        Medium
openssl           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2023-5678        Medium
openssl           3.0.11-1~deb12u2   3.0.13-1~deb12u1   deb     CVE-2023-6237        Unknown
urllib3           2.2.1              2.2.2              python  GHSA-34jh-p97f-mpxf  Medium
```
### After (`1.2.3`) 🎉 
```bash
➜  ~ grype docker.io/airbyte/python-connector-base:1.2.3 --only-fixed
 ✔ Vulnerability DB                [no update available]
 ✔ Loaded image                                                                                                                                               index.docker.io/airbyte/python-connector-base:1.2.3
 ✔ Parsed image                                                                                                                           sha256:55bb33502c977c81e961d5a2864df98045d688d7084eaa64fbb2c7db60c36fcb
 ✔ Cataloged contents                                                                                                                            4d38ca7e933b74c3a1b412dd7e7fe29ddbed79d4b568e64cc80c49d9b72b5ff8
   ├── ✔ Packages                        [234 packages]
   ├── ✔ File digests                    [3,728 files]
   ├── ✔ File metadata                   [3,728 locations]
   └── ✔ Executables                     [966 executables]
 ✔ Scanned for vulnerabilities     [0 vulnerability matches]
   ├── by severity: 3 critical, 14 high, 30 medium, 12 low, 115 negligible (29 unknown)
   └── by status:   0 fixed, 203 not-fixed, 203 ignored
No vulnerabilities found
```

## User Impact
This new base image version will start being used by connectors on the next `up-to-date` run.

### CODE REVIEW

**Comment by alafanechere** [comment]
* **#41057** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/41057?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by natikgadzhi — APPROVED** [review]
<3

**Review by clnoll — APPROVED** [review]
LGTM!

File: airbyte-ci/connectors/base_images/base_images/python/bases.py [hunk 1]
```diff
@@ -100,8 +100,10 @@ def get_container(self, platform: dagger.Platform) -> dagger.Container:
             .with_env_variable("POETRY_VIRTUALENVS_IN_PROJECT", "false")
             .with_env_variable("POETRY_NO_INTERACTION", "1")
             .with_exec(["pip", "install", "poetry==1.6.1"], skip_entrypoint=True)
+            # Upgrade system packages
+            .with_exec(["sh", "-c", "apt-get update && apt-get upgrade -y && apt-get dist-upgrade -y && apt-get clean"])
             # Install socat 1.7.4.4
-            .with_exec(["sh", "-c", "apt update && apt-get install -y socat=1.7.4.4-2"])
+            .with_exec(["sh", "-c", "apt-get install -y socat=1.7.4.4-2"])
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-ci/connectors/base_images/base_images/python/bases.py:106]: We're not dropping it, I just install it in a `with_exec` just below, to seprate "upgrades" from "installs".

**Pull request merged**